File: in_Java/0463 Island Perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int n = 0;
        int over = 0;
        int row = grid.length;
        int col = grid[0].length;
        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                if(grid[i][j] == 1){
                    n++;
                    if(i != 0 && grid[i-1][j] == 1) over++;
                    if(i != row-1 && grid[i+1][j] == 1) over++;
                    if(j != 0 && grid[i][j-1] == 1) over++;
                    if(j != col-1 && grid[i][j+1] == 1) over++;
                }                
            }
        }
        return 4 * n - over;
    }
}

File: in_Python/0463 Island Perimeter.py
class Solution(object):
    def helper(self,i,j,row,col,grid):
        c = 4
        if i-1 >= 0 and grid[i-1][j] == 1:
            c -= 1
        if j-1 >= 0 and grid[i][j-1] == 1:
            c -= 1
        if i+1 < row and grid[i+1][j] == 1:
            c -= 1
        if j+1 < col and grid[i][j+1] == 1:
            c -= 1
        return c
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        res = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    res += self.helper(i,j,len(grid),len(grid[0]),grid)
        return res
        